Assets/Script/InputManager.cs

```diff
@@ -0,0 +1,23 @@
+using UnityEngine;
+
+public class InputManager : MonoBehaviour
+{
+    private InputMap inputMap;
+
+    public void Init(GameObject player)
```

Comment:
Consider using interfaces in these cases instead of passing the GameObject.

e.g:
Create a new interface:
```
public interface IPlayerAction
{
    public void Fire();

    public void Move(float move);

    public void CancelMove();

    public void Rotate(float rotation);

    public void CancelRotate();
}
```

Then modify your Init method to receive and use the interface:
```
    public void Init(IPlayerAction player)
    {
        inputMap = new InputMap();

        inputMap.Gameloop.Fire.performed += context => player.Fire();

        inputMap.Gameloop.Move.started += context => player.Move(inputMap.Gameloop.Move.ReadValue<float>());
        inputMap.Gameloop.Move.canceled += context => player.CancelMove();

        inputMap.Gameloop.Rotate.started += context => player.Rotate(inputMap.Gameloop.Rotate.ReadValue<float>());
        inputMap.Gameloop.Rotate.canceled += context => player.CancelRotate();

        inputMap.Gameloop.Fire.Enable();
        inputMap.Gameloop.Move.Enable();
        inputMap.Gameloop.Rotate.Enable();
    }
```

Create a new class and add that class to your prefab "Tank_Base"
```
public class PlayerGeneric : MonoBehaviour, IPlayerAction
{
    // add these references in the editor, so you avoid using GetComponent when you need them
    [SerializeField] Shoter shooter;
    [SerializeField] Mover mover;
    [SerializeField] PlayerRotator rotator;

    public void CancelMove()
    {
        mover.onPressButton.Invoke(false);
    }

    public void CancelRotate()
    {
        rotator.onPressButton.Invoke(false);
    }

    public void Fire()
    {
        shooter.onShot.Invoke();
    }

    public void Move(float move)
    {
        mover.onMove.Invoke(move);
    }

    public void Rotate(float rotation)
    {
        rotator.onRotate.Invoke(rotation);
    }
}

```


In your PlayerManager instead of
`[SerializeField] private GameObject playerGeneric;
you can use
`[SerializeField] private PlayerGeneric playerGeneric;

Then instead of instatiating it like this:
`GameObject player = Instantiate(playerGeneric);
you use
`PlayerGeneric player = Instantiate(playerGeneric);

This will leave you with the PlayerGeneric you can pass to the InputManager.Init.

It is not considered a good practice to use GetComponent too much.